src/structure.ts

```diff
@@ -127,23 +130,25 @@ const isComponent = (value: any, warn = false): value is Component => {
     return true;
 };
 
+type CompType = { [key: string]: Component };
+
 /**
  * Validate that a set of values are {@link Component} objects
  */
-export const validateComponents = <Ctx extends Context = Context, Req extends Request = Request, Sentry extends Toucan | undefined = undefined>(cmps: any[], warn = false) =>
+export const validateComponents = <Ctx extends Context = Context, Req extends Request = Request, Sentry extends Toucan | undefined = undefined>(cmps: unknown[], warn = false) =>
     cmps.reduce((acc, cmp) => {
         if (!isComponent(cmp, warn)) return acc;
 
         // Check the component doesn't already exist
-        if (acc[cmp.name]) {
+        if ((acc as CompType)[cmp.name]) {
             if (warn)
                 console.warn(`Component ${cmp.name} already exists`);
             return acc;
         }
 
         // Add the component
         return {
-            ...acc,
+            ...(acc as CompType),
```

Comment:
Why are we casting here? If we need this to be correctly typed, why not just type the reduce itself properly?